### PR TITLE
Add background processes

### PR DESCRIPTION
**Pull request by JanLikar**
Use `devenv up -d` to run honcho with `nohup`.

`devenv down` and `devenv logs` are also included and are analog to their `docker-compose` counterparts.

### CODE REVIEW

**Comment by JanLikar** [comment]
This is first and foremost a PoC, so feel free to voice any concerns you might have, I'm more than happy to address them :)

File: src/devenv.nix [hunk 1]
```diff
@@ -77,12 +77,32 @@ pkgs.writeScriptBin "devenv" ''
...
       fi
-      ;;
+    ;;
+    down)
+      if [ -r "$DEVENV_DIR/honcho.pid" ]; then
+        echo "Stopping processes ..." 1>&2
+        echo "" 1>&2
+        kill $(cat "$DEVENV_DIR/honcho.pid")
+        rm "$DEVENV_DIR/honcho.pid"
+      fi;
```
Comment (shyim) [inline src/devenv.nix:101]: Maybe info that no background process could be found?

Comment (JanLikar) [reply src/devenv.nix:101]: That's a good point, I will add it.

**Review by shyim — APPROVED** [review]

**Comment by domenkozar** [comment]
I'm still not sure about this one. 

Could you explain what kind of workflow do you have that requires this interface?

**Comment by shyim** [comment]
I would like to call that also in Gitpod to have one shell.

devenv up -d
devenv shell

and let the user work in the shell and don't have to care about the running services

**Comment by JanLikar** [comment]
I guess so, yes

**Comment by thenonameguy** [comment]
I recommend creating a prototype using https://github.com/NixOS/nixpkgs/blob/e66309ddcce8fc316b9addeabd11bf2187fa2eb9/pkgs/tools/system/daemonize/default.nix. 
This works by wrapping arbitrary programs and turns them into daemon processes.
Should be what we need without relying on the programs to do this themselves.

**Comment by thenonameguy** [comment]
Started discussion about upstreaming this feature to `process-compose` so we don't need the above comment:
https://github.com/F1bonacc1/process-compose/discussions/35

This way not only the honcho implementation could fulfill the new interface.

**Comment by Nebucatnetzer** [comment]
Another use case is bootstrapping the project.
E.g. I start Postgres and Django in the background and then I run some commands to run migrations and insert fixtures.
Afterwards I open the URL for the dev server in the browser.
The dev then can start developing and if he needs to reattach to the log view.
I'm currently doing this with overmind.

**Comment by domenkozar** [comment]
This is now implemented in #745

**Pull request closed without merging**